### PR TITLE
citra-qt: Improve Title Bar display

### PR DESCRIPTION
**Pull request by zhaowenlan1779**
follow-up of #3783.
Addressed @jroweboy's comments

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3804)

### CODE REVIEW

**Comment by valentinvanelslande** [comment]
The game title is removed from the title bar after changing the language if a game is running.

**Comment by zhaowenlan1779** [comment]
@valentinvanelslande I see

**Comment by zhaowenlan1779** [comment]
@valentinvanelslande Is the issue fixed now? Please help me test it.

**Comment by valentinvanelslande** [comment]
@zhaowenlan1779 yes

**Comment by zhaowenlan1779** [comment]
I guess you will not see An effect In Canary because of the mingw pr

**Comment by zhaowenlan1779** [comment]
I guess you will not see an effect In Canary because of the mingw pr

**Comment by zhaowenlan1779** [comment]
Rebased

File: src/common/CMakeLists.txt [hunk 1]
```diff
@@ -16,10 +19,21 @@ if ($ENV{CI})
       string(SUBSTRING ${WORD} 0 1 FIRST_LETTER)
       string(SUBSTRING ${WORD} 1 -1 REMAINDER)
       string(TOUPPER ${FIRST_LETTER} FIRST_LETTER)
-      # this leaves a trailing space on the last word, but we actually want that
-      # because of how it's styled in the title bar.
-      set(REPO_NAME "${REPO_NAME}${FIRST_LETTER}${REMAINDER} ")
+      set(REPO_NAME "${REPO_NAME}${FIRST_LETTER}${REMAINDER}")
     endforeach()
+    if(BUILD_TAG)
```
Comment (baka0815) [inline src/common/CMakeLists.txt]: Everywhere else there is a space between "if" and the opening bracket.

**Comment by zhaowenlan1779** [comment]
@baka0815 see my 2nd commit

**Comment by zhaowenlan1779** [comment]
@baka0815 fixed.

File: src/citra/emu_window/emu_window_sdl2.cpp [hunk 2]
```diff
@@ -100,8 +100,8 @@ EmuWindow_SDL2::EmuWindow_SDL2(bool fullscreen) {
     SDL_GL_SetAttribute(SDL_GL_BLUE_SIZE, 8);
     SDL_GL_SetAttribute(SDL_GL_ALPHA_SIZE, 0);
 
-    std::string window_title = Common::StringFromFormat("Citra %s| %s-%s ", Common::g_build_name,
-                                                        Common::g_scm_branch, Common::g_scm_desc);
+    std::string window_title = Common::StringFromFormat(
+        "Citra %s| %s-%s ", Common::g_build_fullname, Common::g_scm_branch, Common::g_scm_desc);
```
[the remark this answers is not in the document]
Comment (zhaowenlan1779) [reply src/citra/emu_window/emu_window_sdl2.cpp]: I don't think sdl2 frontend needs to be changed :)
I am just wanting to have a more friendly user frontend, and obviously most users will use citra-qt

File: src/common/CMakeLists.txt [hunk 3]
```diff
@@ -16,10 +19,21 @@ if ($ENV{CI})
...
+        set(BUILD_VERSION ${CMAKE_MATCH_1})
+      endif()
+      if (BUILD_VERSION)
+        # This leaves a trailing space on the last word, but we actually want that
+        # because of how it's styled in the title bar.
+        set(BUILD_FULLNAME "${REPO_NAME} #${BUILD_VERSION} ")
+      else()
+        set(BUILD_FULLNAME "")
+      endif()
+    endif()
```
Comment (wwylele) [inline src/common/CMakeLists.txt]: I prefer to move these string building logic to the actual c++ code. It is easier to debug. In fact, currently due to the problem in #3706 BUILD_TAG is never set and the whole logic is skipped here, leaving an empty string in the title.

Comment (zhaowenlan1779) [reply src/common/CMakeLists.txt]: I found it better to set it in CMakeLists.
BUILD_FULLNAME will be a constant. And I'm not happy writing logic in a generated src file. And if you build the string while formatting, there will be much repeated logic

**Pull request merged**